File: src/main/java/supercritical/api/nuclear/fission/components/FuelRod.java
package supercritical.api.nuclear.fission.components;

import lombok.Getter;
import lombok.Setter;
import net.minecraft.item.ItemStack;
import supercritical.api.nuclear.fission.IFissionFuelStats;

public class FuelRod extends ReactorComponent {

    @Getter
    private IFissionFuelStats fuel;
    @Getter
    @Setter
    private double weight = 1;
    @Getter
    @Setter
    private double thermalProportion = 0;

    public FuelRod(double maxTemperature, double thermalConductivity, IFissionFuelStats fuel, double mass) {
        super(0, maxTemperature, thermalConductivity, mass, true);
        this.fuel = fuel;
    }

    public double getDuration() {
        return fuel.getDuration();
    }

    public double getNeutronGenerationTime() {
        return fuel.getNeutronGenerationTime();
    }

    public void setFuel(IFissionFuelStats property) {
        this.fuel = property;
        this.maxTemperature = property.getMaxTemperature();
    }

    public ItemStack getDepletedFuel() {
        return fuel.getDepletedFuel(thermalProportion);
    }

}

File: src/main/kotlin/supercritical/api/nuclear/fission/components/FuelRod.kt
package supercritical.api.nuclear.fission.components

import net.minecraft.world.item.ItemStack
import supercritical.api.nuclear.fission.IFissionFuelStats

class FuelRod(maxTemperature: Double, thermalConductivity: Double, private var fuel: IFissionFuelStats, mass: Double) :
    ReactorComponent(0.0, maxTemperature, thermalConductivity, mass, true) {
    var weight: Double = 1.0
    var thermalProportion: Double = 0.0

    val duration: Double
        get() = fuel.duration.toDouble()

    val neutronGenerationTime: Double
        get() = fuel.neutronGenerationTime

    fun setFuel(fuel: IFissionFuelStats) {
        this.fuel = fuel
        this.maxTemperature = fuel.maxTemperature.toDouble()
    }

    val depletedFuel: ItemStack?
        get() = fuel.getDepletedFuel(thermalProportion)

    fun getFuel(): IFissionFuelStats {
        return fuel
    }
}
